### ISSUE TITLE
'Kernel dies, restarting' when calling numpy functions 

### ISSUE THREAD

**Issue by andreyvgritsenko** [keras-team/keras#12346]
I train a Keras model and want to perform some matrix operation on the output (e.g. numpy.matmul()). However, each time I try to use any of numpy functions (like, matmul(), dot(), concatenate()), the IPython kernel dies and restarts. I'm using Spyder IDE on Mac OS X, both Keras, Tensorflow, and numpy are up-to-date. 
Please help me identify the culprit! 

Additional details: 
- The problem occurs even if I use numpy functions with newly generated data, instead of the model outputs (see script below)
- Numpy functions do not raise any errors if used before the model training 
- Using tensorflow.matmul instead of numpy.matmul helps to avoid the problem, but it is not a solution as in the end I need to use more than one numpy function

Here's the script I'm running:
`from keras.layers import Input, Dense, Conv2D, MaxPooling2D,Flatten`
`from keras.models import Model`
`import numpy as np`
`import tensorflow as tf`
`import keras.utils as kutils`
`n_y = 10`
`mnist = tf.keras.datasets.mnist`
`(X_train, y_train), (X_test, y_test) = mnist.load_data()`
`y_train = kutils.to_categorical(y_train, n_y)`
`y_test = kutils.to_categorical(y_test, n_y)`
`X_train = X_train.reshape(X_train.shape[0], 28, 28,1)`
`X_test = X_test.reshape(X_test.shape[0], 28, 28,1)`
`channels = 512`
`X = Input(shape=(28,28,1)) # size of MNIST images`
`y = Input(shape=(n_y,))`
`conv1 = Conv2D(32, (3, 3), activation='relu')(X)`
`pool1 = MaxPooling2D((2, 2), strides=(2, 2), name='block1_pool')(conv1)`
`flatten = Flatten()(pool1)`
`h_1 = Dense(64, activation='relu')(flatten)`
`h_2 = Dense(10, activation='softmax')(h_1)`
`model = Model(inputs=[X,y], outputs=h_2)`
`model.summary()`
`model.compile(loss='mean_squared_error', optimizer='adam',metrics=['accuracy'])`
`model.fit([X_train, y_train],y_train, batch_size=channels, epochs=1, validation_split=0.1)`
`np.matmul(np.array([[1, 0],[0, 1]],dtype=np.float64),np.array([[2, 3],[4, 6]],dtype=np.float64))`

**Comment by andreyvgritsenko** [comment]
During one of the trials the kernel didn't restart automatically and I got the following error message:
"OMP: Error #15: Initializing libiomp5.dylib, but found libiomp5.dylib already initialized.
OMP: Hint: This means that multiple copies of the OpenMP runtime have been linked into the program. That is dangerous, since it can degrade performance or cause incorrect results. The best thing to do is to ensure that only a single OpenMP runtime is linked into the process, e.g. by avoiding static linking of the OpenMP runtime in any library. As an unsafe, unsupported, undocumented workaround you can set the environment variable KMP_DUPLICATE_LIB_OK=TRUE to allow the program to continue to execute, but that may cause crashes or silently produce incorrect results. For more information, please see http://www.intel.com/software/products/support/." 

After some googling I found the thread [https://github.com/dmlc/xgboost/issues/1715#issuecomment-438924721](url) and the issue was solved with 
`conda install nomkl`
However, now some computations take much longer than before. Any suggestions on how to fix the problem without slowdowns or any drawbacks are appreciated!
